prototype/distribution/launcher/src/main/java/org/eclipse/sensinact/gateway/launcher/ConfigurationManager.java

```diff
@@ -63,7 +66,10 @@ public class ConfigurationManager {
 
     ExecutorService executor;
 
-    private Path configFile;
+    Path configFile;
+
+    private boolean expectedInterruption = false;
```

Comment:
This needs to be volatile (or better yet an AtomicBoolean) as the signal comes from a different thread and (as far as I am aware) Thread.interrupt() doesn't trigger a memory barrier and force a memory update in either thread.

Reply:
OK. Using an AtomicBoolean is better as we can do a getAndSet while handling the interruption